pyanaconda/payload/__init__.py

```diff
@@ -992,6 +1012,13 @@ def __init__(self, data):
     def preInstall(self):
         super(PackagePayload, self).preInstall()
 
+        # remove any disabled repos so they don't appear in kickstart
+        #  TODO: fixup pykickstart so that it doesn't write out disabled repos
+        #        then this block becomes redundant.
+        for repo in self.disabledRepos:
+            log.debug("Removing disabled repo %s", repo.name)
+            self.removeRepo(repo.name)
+
```

Comment:
Interesting I didn't know this is happening and basically you are correct it doesn't give sense to have disabled repo in an output kickstart file. 
However, you should remove the `TODO` and fix it in pyanaconda/kickstart.py the enabled part is not part of kickstart project but Anaconda's extension. Basically you need to add the `__str__()` method to the `RepoData` class and exclude disabled repositories there.

Reply:
Ahh, that's where that's happening.  I'll get it fixed.